### DISCUSSION TITLE
Examples of creating aggregate functions

### DISCUSSION THREAD

**Discussion by isNotOkay** [rusqlite/rusqlite#1344] [Questions] [answered]
Hello,

Are there any good examples of creating aggregate functions? 

We would like to create a loadable extension with both custom **scalar** and **aggregate** functions as described in the following blog post:
https://ricardoanderegg.com/posts/extending-sqlite-with-rust/

**Comment by gwenn** [comment, marked as answer]
Currently, rusqlite doesn't support loadable extension as shared library (see #910) but `sqlite3_auto_extension` should be fine.
And there are examples here:
https://github.com/rusqlite/rusqlite/blob/38bf316601defcc8717e459c6a80821aea39aa97/src/functions.rs#L949-L1027

**Reply by isNotOkay** [reply]
Thanks for the quick reply! 

We are using [SQLiteStudio](https://sqlitestudio.pl/) and are looking for a way to add custom scalar/aggregate functions via loadable extensions. Do you think this could be achieved with the usage of _sqlite3_auto_extension_ somehow?

**Reply by gwenn** [reply]
I don't think so, you need to follow these instructions from https://ricardoanderegg.com/posts/extending-sqlite-with-rust/ :
> Using [this rusqlite fork from Genomicsplc](https://github.com/Genomicsplc/rusqlite/tree/loadable-extensions). For more context why this is needed, see [#910](https://github.com/rusqlite/rusqlite/pull/910).
UPDATE: I’ve created a new fork which merges the changes in that PR with the current latest rusqlite release (v0.27.0) [here](https://github.com/litements/rusqlite/tree/loadable-extensions-release-2).

or any alternative like [sqlite-loadable](https://crates.io/crates/sqlite-loadable)
or adapt something like [this](https://github.com/lukaslueg/sqlite3_hyperminhash/blob/master/src/shim.c).

We hope that, one day, `rusqlite` will support generating loadable extension directly...

**Reply by isNotOkay** [reply]
Thanks for the links👍. That would be great!

**Comment by gwenn** [comment]
There is also [sqlite3_ext](https://github.com/CGamesPlay/sqlite3_ext).
